pkg/asset/machines/worker.go

```diff
@@ -405,7 +406,17 @@ func (w *Worker) Generate(dependencies asset.Parents) error {
 			}
 
 			pool.Platform.Azure = &mpool
-			sets, err := azure.MachineSets(clusterID.InfraID, ic, &pool, string(*rhcosImage), "worker", workerUserDataSecretName)
+			ssn, err := icazure.GetSession(installConfig.Azure.CloudName, installConfig.Azure.ARMEndpoint)
+			if err != nil {
+				return err
+			}
```

Comment:
Do we need a new session? It looks like there is already a session in this context